blocks/v2-hero/v2-hero.js

```diff
@@ -0,0 +1,98 @@
+import {
+  createElement,
+  isVideoLink,
+  removeEmptyTags,
+} from '../../scripts/scripts.js';
+
+const checkVideoTime = (event) => {
+  const video = event.target;
+  if (video.currentTime > 3) {
+    video.pause();
+    video.removeEventListener('timeupdate', checkVideoTime);
+  }
+};
+
+export default async function decorate(block) {
+  const blockName = 'v2-hero';
+
+  const picture = block.querySelector('picture');
+  const link = block.querySelector('a');
+  const isVideo = link ? isVideoLink(link) : false;
+  if (isVideo) {
+    const video = createElement('video', [`${blockName}__video`]);
+    video.muted = true;
+    video.autoplay = true;
+
+    const source = createElement('source', '', {
+      src: link.getAttribute('href'),
+      type: 'video/mp4',
+    });
+    video.appendChild(source);
+    block.prepend(video);
+    link.remove();
+
+    // prevent video to reproduce more than 3s
+    video.addEventListener('timeupdate', checkVideoTime);
+  }
+
+  if (picture) {
+    const img = picture.querySelector('img');
+    img.classList.add(`${blockName}__image`);
+    if (picture.parentElement.tagName === 'P') {
+      picture.parentElement.remove();
+    }
+    block.prepend(picture);
+  }
+
+  const contentWrapper = block.querySelector(':scope > div');
+  contentWrapper.classList.add(`${blockName}__content-wrapper`);
+
+  // add Hero text variants to the corresponding content
+  const variantClasses = ['centered', 'left', 'bottom', 'dark'];
```

Comment:
Could we declare the `variantClasses` at the top of the `decorate` function? Thanks for that it will be easier to check what variants the block support. It is dev friendly 😄

Reply:
done